File: src/main/java/de/synyx/google/calendar/api/model/Event.java
package de.synyx.google.calendar.api.model;

import java.time.Duration;
import java.time.ZonedDateTime;
import java.util.Optional;
import java.util.stream.Stream;

/**
 * @author clausen - clausen@synyx.de
 */
public interface Event {

    public String name ();

    public String description ();

    public String location ();

    public Stream<Attendee> attendees ();

    public ZonedDateTime start ();

    public Optional<ZonedDateTime> end ();

    public default Optional<Duration> duration () {
        return end ().map (end -> Duration.between (start (), end));
    }

}

File: modules/core/src/main/kotlin/de/synyx/google/calendar/core/api/model/Event.kt
package de.synyx.google.calendar.core.api.model

import org.joda.time.Duration
import org.joda.time.Instant

/**
 * @author clausen - clausen@synyx.de
 */
interface Event {

    fun id () : String

    fun title () : String

    fun description () : String

    fun location () : String

    fun creator () : Attendee

    fun attendees () : Collection<Attendee>

    fun start () : Instant

    fun end () : Instant?

    fun duration () : Duration = Duration (start (), end ())

}
